File: app/src/main/java/com/example/bookapp/AppUtilities.java
package com.example.bookapp;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.json.JSONException;
import org.json.JSONObject;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class AppUtilities {
    private static Retrofit retrofit;

    public static Retrofit getRetrofit() {
        if (retrofit == null) {
            Retrofit.Builder builder = new Retrofit.Builder().baseUrl("http://139.162.192.88");
            builder.addConverterFactory(GsonConverterFactory.create());
            retrofit = builder.build();
        }
        return retrofit;
    }


    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager
                = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
        return activeNetworkInfo != null && activeNetworkInfo.isConnected();
    }

    /**
     * This method uses a regex with the matches() method in
     * order to determine if the email address is valid or
     * not
     *
     * @param email
     * @return
     */
    public static boolean isEmailValid(String email) {
        return email.matches("[a-zA-Z0-9]+@[a-z]+\\.[a-z]+");
    }

    public static String getDateString() {
        Date date = Calendar.getInstance().getTime();
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd  H:m:s");
        return simpleDateFormat.format(date);
    }

    public static String convertUnixTimeToStringDate(long unixTime) {
        Date date = new Date(unixTime * 1000);
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
        return simpleDateFormat.format(date);
    }

    public static boolean isResponsePositive(JSONObject response) {
        try {
           return response.getInt("responseCode") == 200;
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return false;
    }
}

File: app/src/main/java/com/example/bookapp/AppUtilities.kt
package com.example.bookapp

import android.app.Activity
import android.content.Context
import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable
import android.net.ConnectivityManager
import android.net.Uri
import android.util.Base64
import android.util.DisplayMetrics
import java.io.ByteArrayOutputStream
import java.io.InputStream

fun Context.getScreenWidth(): Int {
    return this.resources.displayMetrics.widthPixels
}


fun Context.getConnectivityManager() = this.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

/**
 * This method uses a regex with the matches() method in
 * order to determine if the email address is valid or
 * not
 *
 * @param email
 * @return
 */

fun String.isEmail(): Boolean {
    val regex = "[a-zA-Z0-9]+@[a-z]+\\.[a-z]+"
    return this.matches(regex.toRegex())
}


fun Uri.toDrawable(context: Context): Drawable {
    val inputStream: InputStream? = context.contentResolver.openInputStream(this)
    return Drawable.createFromStream(inputStream, path.toString())
}


fun Drawable.toBase64(): String {
    val bitmap = (this as BitmapDrawable).bitmap
    val byteArrayOutputStream = ByteArrayOutputStream()
    bitmap.compress(Bitmap.CompressFormat.JPEG, 70, byteArrayOutputStream)
    val byteArray = byteArrayOutputStream.toByteArray()
    return Base64.encodeToString(byteArray, Base64.DEFAULT);
}

